pkg/cmd/cli/cmd/set/buildsecret.go

```diff
@@ -0,0 +1,283 @@
+package set
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	buildSecretLong = `
+Set or remove a build secret on a build config
+
+A build config can reference a secret to push or pull images from private registries or
+to access private source repositories.
+
+Specify the type of secret being set by using the --push, --pull, or --source flags.
+A secret reference can be removed by using --remove flag.
+
+A label selector may be specified with the --selector flag to select the build configs
+on which to set or remove secrets. Alternatively, all build configs in the namespace can
+be selected with the --all flag.
+`
+
+	buildSecretExample = `  # Clear push secret on a build config
+  %[1]s build-secret --push --remove bc/mybuild
+
+  # Set the pull secret on a build config
+  %[1]s build-secret --pull bc/mybuild mysecret
+
+  # Set the source secret on a set of build configs matching a selector
+  %[1]s build-secret --source -l app=myapp gitsecret
```

Comment:
Might be worth adding an example which sets for more than one secret type, it seems to be a common use case.